### PR TITLE
Argument Relative Assignment Fix

### PR DESCRIPTION
**Pull request by RobertBColton**
This fixes an issue with emake's drag and drop conversion that was causing `argument_relative` not to be assigned correctly. It specifically was making TKG's Key to Success game have collision bugs when run from emake. The problem was that JDI has a setting to treat `=` assignment as `==` comparison like old GM, which interferes with the behavior of the converted drag and drop code. LGM plugin solved this problem by using the `:=` assignment operator, but everybody on Discord pushed me to just make it its own separate statement so it's more human-readable. With this change, TKG's Key to Success works perfectly from the command line.

### CODE REVIEW

File: CommandLine/libEGM/action.cpp [hunk 1]
```diff
@@ -116,6 +116,8 @@ std::string Actions2Code(const std::vector< buffers::resources::Action >& action
       case ActionKind::ACT_NORMAL:
         if (action.exe_type() == ActionExecution::EXEC_NONE) break;
 
+        code += "argument_relative = " + std::to_string(action.relative()) + ";\n";
```
Comment (JoshDreamland) [inline CommandLine/libEGM/action.cpp:119]: Not a compelling reason not to pre-concatenate the string.
```suggestion
        code += action.relative() ? "argument_relative = true;\n" : "argument_relative = false;\n";
```

**Comment by JoshDreamland** [comment]
This also has the possibility to introduce regressions. Does Key to Success feature a "Question" action that is relative? You might have broken them.

1. Do we have tests for *any* of this? Actual tests, not just running one game.
2. Does this change (or a similar change) also need to be made to LGM's Action generator?

**Comment by RobertBColton** [comment]
We don't have drag and drop tests yet, no, but they are needed. This change does not need to be made to LGM's action generator, no, because it uses the `:=` assignment operator from GML, because the author knew well enough that JDI converts `=` to `==` in parentheses. That was the mistake I made, I didn't understand why `:=` was being used in lgmplugin, and thought it looked silly.

**Comment by JoshDreamland** [comment]
IIRC, the result of an `action_if_*` function is used directly in an if-statement when emitting code. That's why we inlined the assignment.

**Comment by RobertBColton** [comment]
I still don't know the answer to that, but I do believe it was specifically a question action broken in his game. Also, while lgmplugin used the result inline, emake uses the `__if__` fix for "applies to" fix like GMS.

**Comment by fundies** [comment]
@RobertBColton when you merge this and #1989? I would like emake to work proper pls

**Comment by RobertBColton** [comment]
Soon

**Comment by RobertBColton** [comment]
@JoshDreamland I had to double check what you were saying here and I couldn't find it to be the case. I don't think I originally understood you but I do now in that you were concerned any of the `action_if` question actions take `argument_relative` as a first parameter instead of using the global `argument_relative` variable. I could not find a single action if function in ENIGMA that has relative as its first parameter, so that could have just been a mistake in the original LGM plugin I copied over.

If you go use ActionMaker to open the GM action libraries, you can see several questions use argument_relative but don't have it as a parameter. I don't even see an option to have it as a parameter.
https://robertbcolton.github.io/ActionMaker/

Also if we inspect the conversion code some more, we see the function name/call was inside the parentheses _after_ where the assignment I removed was.
https://github.com/enigma-dev/enigma-dev/blob/95209cfffbbd395877444d50378c473b459fe13f/CommandLine/libEGM/action.cpp#L139

So really I think you just misread and this should be fine.

**Pull request merged**